src/status_im/ui/screens/events.cljs

```diff
@@ -215,17 +215,14 @@
    :on-accept           handle-reset-data})
 
 (defn initialize-views [{{:accounts/keys [accounts] :as db} :db}]
-  {:db                                  (if (empty? accounts)
-                                          (assoc db :view-id :intro :navigation-stack (list :intro))
-                                          (let [{:keys [address photo-path name]} (first (sort-by :last-sign-in > (vals accounts)))]
-                                            (-> db
-                                                (assoc :view-id :login
-                                                       :navigation-stack (list :login))
-                                                (update :accounts/login assoc
-                                                        :address address
-                                                        :photo-path photo-path
-                                                        :name name))))
-   :handle-initial-push-notification-fx db})
+  (let [{:keys [address photo-path name]} (first (sort-by :last-sign-in > (vals accounts)))]
+    (cond-> {:handle-initial-push-notification-fx db}
+      ;; if there is nothing, let's just show the intro
+      (empty? address)       (assoc :db (assoc db :view-id :intro :navigation-stack (list :intro)))
+                                 ;; otherwise, we pre-fill the nav stack
+      (not (empty? address)) (-> (assoc :db (assoc db :navigation-stack (list)))
+                                 ;; and trying to open login
+                                 (assoc :dispatch [:open-login address photo-path name])))))
```

Comment:
That pattern is unfortunately very wrong (unnecessary chaining of events) and originally was everywhere in the app.
It's now much better and recently refactored code (like for example chat input/sending/receiving) doesn't suffer from it at all.
General very simple rule would be - only dispatch as consequence of external action, be it some user input (on-click/on-change callback in view), signal in asynchronous process (callback from http call), etc. 
Do not do it where no asynchronous behavior is needed and when simple function call is sufficient.
By doing it, we are unnecessary bringing asynchronous concerns and potential race condition into the game, because not only are we writing side-effect code (event dispatch is side-effect, impure action) where it could be pure functional, we are also writing asynchronous code for no good reason other then "writing it that way is convenient and we don't have to think about return values", which is very poor excuse, given that we choosed  functional language as our tool :).
Another bad thing is that events are global and so you can quickly create circular dependencies that way, with no sane hierarchy of namespaces and their dependencies (totally intermingled quagmire) which would otherwise be forbidden by the cljs compiler.

To provide some useful advice and not only mad ranting, here is my suggestion how it could implemented in functional way:
```clj
(let [default-effects {:handle-initial-push-notification-fx db}]
  (if (empty? address)
    (handlers-macro/merge-fx cofx 
                             default-effects
                             (navigation/navigate-to-clean :intro))
    (handlers-macro/merge-fx cofx
                             default-effects
                             (model/open-login last-account))))
```